subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -71,69 +71,49 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv);
-#if defined(CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET)
-			BT_MESH_ADV_TAG_BIT_RELAY |
-#endif /* CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET */
-#if defined(CONFIG_BT_MESH_PB_ADV)
-			BT_MESH_ADV_TAG_BIT_PROV |
-#endif /* CONFIG_BT_MESH_PB_ADV */
-			BT_MESH_ADV_TAG_BIT_LOCAL
-		),
-		.work = Z_WORK_INITIALIZER(send_pending_adv),
-	},
-#if CONFIG_BT_MESH_RELAY_ADV_SETS
-	[1 ... CONFIG_BT_MESH_RELAY_ADV_SETS] = {
-		.tags = (
-#if defined(CONFIG_BT_MESH_RELAY)
-			BT_MESH_ADV_TAG_BIT_RELAY |
-#endif /* CONFIG_BT_MESH_RELAY */
-#if defined(CONFIG_BT_MESH_PB_ADV_USE_RELAY_SETS)
-			BT_MESH_ADV_TAG_BIT_PROV |
-#endif /* CONFIG_BT_MESH_PB_ADV_USE_RELAY_SETS */
-		0),
-		.work = Z_WORK_INITIALIZER(send_pending_adv),
-	},
-#endif /* CONFIG_BT_MESH_RELAY_ADV_SETS */
-#if defined(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE)
-	{
-		.tags = BT_MESH_ADV_TAG_BIT_FRIEND,
-		.work = Z_WORK_INITIALIZER(send_pending_adv),
-	},
-#endif /* CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE */
-#if defined(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE)
-	{
-		.tags = BT_MESH_ADV_TAG_BIT_PROXY,
+		.tags = (BT_MESH_ADV_TAG_BIT_LOCAL |
+				 (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE)
+					  ? 0 : BT_MESH_ADV_TAG_BIT_FRIEND) |
+				 (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE)
+					  ? 0 : BT_MESH_ADV_TAG_BIT_PROXY) |
+				 (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET)
+					  ? BT_MESH_ADV_TAG_BIT_RELAY : 0) |
+				 (IS_ENABLED(CONFIG_BT_MESH_PB_ADV)
+					  ? BT_MESH_ADV_TAG_BIT_PROV : 0)),
 		.work = Z_WORK_INITIALIZER(send_pending_adv),
 	},
-#endif /* CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE */
+	COND_CODE_0(CONFIG_BT_MESH_RELAY_ADV_SETS,
+		(), ([1 ... CONFIG_BT_MESH_RELAY_ADV_SETS] = {
+			.tags = (
+			  (IS_ENABLED(CONFIG_BT_MESH_RELAY) ?
+				BT_MESH_ADV_TAG_BIT_RELAY : 0) |
+			  (IS_ENABLED(CONFIG_BT_MESH_PB_ADV_USE_RELAY_SETS) ?
+				BT_MESH_ADV_TAG_BIT_PROV : 0)
+			),
+			.work = Z_WORK_INITIALIZER(send_pending_adv),
+		    }))
+	COND_CODE_1(IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE),
+		({  .tags = BT_MESH_ADV_TAG_BIT_FRIEND,
+		    .work = Z_WORK_INITIALIZER(send_pending_adv),
+		 }), ())
+	COND_CODE_1(IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE),
+		({  .tags = BT_MESH_ADV_TAG_BIT_PROXY,
+		    .work = Z_WORK_INITIALIZER(send_pending_adv),
+		 }), ())
 };
 
 BUILD_ASSERT(ARRAY_SIZE(advs) <= CONFIG_BT_EXT_ADV_MAX_ADV_SET,
 	     "Insufficient adv instances");
 
 static inline struct bt_mesh_ext_adv *relay_adv_get(void)
 {
-	if (!!(CONFIG_BT_MESH_RELAY_ADV_SETS)) {
-		return &advs[1];
-	} else {
-		return &advs[0];
-	}
+	return !!(CONFIG_BT_MESH_RELAY_ADV_SETS) ? &advs[1] : &advs[0];
 }
 
 static inline struct bt_mesh_ext_adv *gatt_adv_get(void)
 {
-	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE)) {
-		return &advs[ARRAY_SIZE(advs) - 1];
-	} else {
-		return &advs[0];
-	}
+	return IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE) ?
+			&advs[ARRAY_SIZE(advs) - 1] : &advs[0];
```

Comment:
I know you're a fan of compact code, however I really don't think this improves readability. The C ternary operator is useful in places where you can't use `if ()` however I don't think it improves readability at all (IMO it does the contrary).